src/sandbox/external-resources.js

```diff
@@ -32,13 +39,28 @@ function addJS(resource: string) {
   document.head.appendChild(script);
 }
 
-export function addResource(resource: string, addCSS = addCSS, addJS = addJS) {
+export function addResource(
+  resource: string,
+  addCSS = addCSS,
+  addJS = addJS,
+  getContentType = getContentType
+) {
   const match = resource.match(/\.([^.]*)$/);
 
   if (match && match[1] === 'css') {
     addCSS(resource);
-  } else {
+  } else if (match && match[1] === 'js') {
     addJS(resource);
+  } else {
+    return getContentType(resource).then(contentType => {
+      if (contentType.indexOf('text/css') === 0) {
+        addCSS(resource);
+      } else if (contentType.indexOf('application/javascript') === 0) {
```

Comment:
`application/javascript` is not the only content type for javascript source code, see [here](http://2ality.com/2011/08/javascript-media-type.html).

Reply:
According to my research, [`application/javascript` is official](https://stackoverflow.com/questions/4101394/javascript-mime-type`). We could also add `application/ecmascript`, since it seems like it's in use. The `text/` ones are deprecated. Do you think it's OK to leave them out? The rest seem obscure.

Reply:
I looked at the 'Content-Type' header for the scripts in a bunch of random websites, and the results were all over the place, from `application/javascript`, to `text/javascript`, to `application/x-javascript` etc.

Reply:
OK. In that case, we can probably just go ahead and add the whole list.